### PR TITLE
GC: Backup:  Sharepoint libraries to omit site drive.

### LINKED ISSUE #2072: GC: Backup: SharePoint: Libraries --> Omit Site Pages
Site Pages to be saved as their own data type. Change backup workflow for libraries to omit this drive.

### PR DESCRIPTION
**Pull request by dadams39**
## Description
Branch treats Drive `Site Pages` as a restricted directory. All  `pages` within the directory, will be backed up via the Site Pages API.  This adds an additional call to per collection to obtain the parent `driveName`. 

TODO: Create a Pipeline that distinguishes between SitePages, Libraries, and List for SharePoint
## Does this PR need a docs update or release note?

- [x] :clock1: Yes, but in a later PR
The documentation should state the assumption is that only `.aspx` files are to be found in the s
## Type of change

- [x] :sunflower: Feature

## Issue(s)

* closes #2072<issue>

## Test Plan
- [x] :zap: Unit test

### CODE REVIEW

File: src/internal/connector/onedrive/collections.go [hunk 1]
```diff
@@ -165,6 +166,39 @@ func (c *Collections) UpdateCollections(ctx context.Context, driveID string, ite
 	return nil
 }
 
+func shouldSkipDrive(ctx context.Context, service graph.Servicer, drivePath path.Path, m folderMatcher) bool {
+	shouldSkip := !includePath(ctx, m, drivePath)
+	if shouldSkip {
+		return true
+	}
+
+	return restrictedPath(ctx, service, drivePath)
```
Comment (ryanfkeepers) [inline src/internal/connector/onedrive/collections.go:175]:
```suggestion
	return !includePath(ctx, m, drivePath) || restrictedPath(ctx, service, drivePath)
```

File: src/internal/connector/onedrive/collections.go [hunk 2]
```diff
@@ -165,6 +166,39 @@ func (c *Collections) UpdateCollections(ctx context.Context, driveID string, ite
...
+	if len(folders) < 2 {
+		return false
+	}
+
+	driveObj, err := service.
+		Client().
+		DrivesById(folders[1]).
+		Get(ctx,
+			optionsSingleDrive([]string{"id", "name"}),
+		)
```
Comment (ryanfkeepers) [inline src/internal/connector/onedrive/collections.go:189]: Nit, func params should be inlined, or all multilined.
```suggestion
		Get(ctx, optionsSingleDrive([]string{"id", "name"}))
```

File: src/internal/connector/onedrive/collections.go (hunk 1 above)
Comment (dadams39) [reply src/internal/connector/onedrive/collections.go]: Resolved in [1565052](https://github.com/alcionai/corso/pull/2098/commits/1565052da72e1204d2f07bc9eb309f54bc9e3c3b)

File: src/internal/connector/onedrive/collections.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ const (
 	OneDriveSource
 	SharePointSource
 )
+const restrictedDirectory = "Site Pages"
```
[the remark this answers is not in the document]
Comment (dadams39) [reply src/internal/connector/onedrive/collections.go:27]: Resolved in [1565052](https://github.com/alcionai/corso/pull/2098/commits/1565052da72e1204d2f07bc9eb309f54bc9e3c3b)

**Review by ryanfkeepers — APPROVED** [review]

**Pull request merged**